### DISCUSSION TITLE
Spring boot With mongock.  What could be a better way to access Environment Variables?

### DISCUSSION THREAD

**Discussion by manishpurohit92** [flamingock/mongock#59] [Q&A] [answered]
Is it possible to use @Value Spring annotation with mongock? Or what are all the options

**Comment by manishpurohit92** [comment]
Earlier Mongobee has this method. 

 public Mongobee setSpringEnvironment(Environment environment) {
        this.springEnvironment = environment;
        return this;
    }

**Comment by dieppa** [comment, marked as answer]
Hello @manishpurohit92 ,

Mongock doesn't have support yet for injecting environment variables via @Value annotations. It will be added in version 5 though.

However, you can inject the Environment object itself with something like this:

```java
    @ChangeSet(id = "your-changelog-id", order = "001", author = "mongock")
    public void dataInitializer(@NonLockGuarded(NonLockGuardedType.NONE) Environment environment) {
        // TODO 
    }
```
